src/api/controllers/user-controllers.ts

```diff
@@ -95,47 +97,49 @@ const updateUserProfile = async (req: Request, res: Response, next: NextFunction
 const addFavoriteActivity = async (req: Request, res: Response, next: NextFunction) => {
     /** Get the user firebase id (identifier) and the id of the activity we want to favorite (update) */
     const { identifier } = req.body;
-    var { update } = req.body;
-
-    /** Verify user exists */
-    const userCheck = await userServices.getUserById(identifier.firebase_id);
-
-    /** If user exists */
-    if (userCheck.found && userCheck.user) {
-        /** array to be sent to the update method in the service layer */
-        let favoriteActivitiesOld = userCheck.user.favoriteActivities;
-
-        /** If the user already has a list of favorite activities, then we will modify the current list and send it to the service layer*/
-        if (userCheck.user.favoriteActivities) {
-            const index = favoriteActivitiesOld.findIndex((element) => element == update.favoriteActivities);
-
-            /** If the current activity was not already favorited, then add it to the user's list of favorites*/
-            if (index == -1) {
-                favoriteActivitiesOld.push(update.favoriteActivities);
-            } /** If the current activity was already favorited in the past, then simply un-favorite it*/ else {
-                favoriteActivitiesOld.splice(index, 1);
+    let { update } = req.body;
```

Comment:
const (kinda nitpicking lol)

Reply:
the update variable will be updated later on, that's why it can't be a const I think.It was const originally but when I tried to modify it it gave me a compiler error

Reply:
oh okay. yeah nvm

Reply:
FYI, it's modify at line 127